python/dgl/distributed/dist_dataloader.py

```diff
@@ -0,0 +1,159 @@
+# pylint: disable=global-variable-undefined, invalid-name
+"""Multiprocess dataloader for distributed training"""
+import multiprocessing as mp
+import time
+import traceback
+
+from . import exit_client
+from .rpc_client import get_sampler_pool
+from .. import backend as F
+
+__all__ = ["DistDataLoader"]
+
+
+def call_collate_fn(next_data):
+    """Call collate function"""
+    try:
+        result = DGL_GLOBAL_COLLATE_FN(next_data)
+        DGL_GLOBAL_MP_QUEUE.put(result)
+    except Exception as e:
+        traceback.print_exc()
+        print(e)
+        raise e
+    return 1
+
+
+def init_fn(collate_fn, queue, sig_queue):
+    """Initialize setting collate function and mp.Queue in the subprocess"""
+    global DGL_GLOBAL_COLLATE_FN
+    global DGL_GLOBAL_MP_QUEUE
+    global DGL_SIG_QUEUE
+    DGL_SIG_QUEUE = sig_queue
+    DGL_GLOBAL_MP_QUEUE = queue
+    DGL_GLOBAL_COLLATE_FN = collate_fn
+    time.sleep(1)
+    return 1
+
+
+def _exit():
+    exit_client()
+    time.sleep(1)
+
+
+def enable_mp_debug():
+    """Print multiprocessing debug information"""
+    import logging
+    logger = multiprocessing.log_to_stderr()
+    logger.setLevel(logging.DEBUG)
```

Comment:
how does the logger work? i don't see where is used in the code.

Reply:
It's for debug usage. I just want to keep it in case of debugging in the future

Reply:
you should document it in the comment.